### PR TITLE
Remove RestApiVersion#V_7 references for 9.0.0

### PR DESCRIPTION
**Pull request by drempapis**
This work addresses and removes references to `org elastic search.core.RestApiVersion#V_7` from the search-related code.

[JIRA reference](https://elasticco.atlassian.net/browse/ES-9453)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java [hunk 1]
```diff
@@ -28,10 +25,6 @@ public class CommonTermsQueryBuilder extends AbstractQueryBuilder<CommonTermsQue
     public static final String COMMON_TERMS_QUERY_DEPRECATION_MSG = "Common Terms Query usage is not supported. "
         + "Use [match] query which can efficiently skip blocks of documents if the total number of hits is not tracked.";
 
-    @UpdateForV9(owner = UpdateForV9.Owner.SEARCH_RELEVANCE) // v7 REST API no longer exists: eliminate ref to RestApiVersion.V_7
-    public static ParseField NAME_V7 = new ParseField("common").withAllDeprecated(COMMON_TERMS_QUERY_DEPRECATION_MSG)
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java:32]: Looks like there is more to do around this cleanup. Can you do some digging to figure it out? I think we may be able to remove the query builder entirely.

**Review by javanna — COMMENTED** [review]
I left some comments, thanks for taking this on!

File: server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java (hunk 1 above)
Comment (drempapis) [reply server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java:32]: That's right, it was added in this pr https://github.com/elastic/elasticsearch/pull/75896

File: server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java [hunk 2]
```diff
@@ -281,14 +267,11 @@ public static void readMultiLineFormat(
                             allowNoIndices = value;
                         } else if ("ignore_throttled".equals(entry.getKey()) || "ignoreThrottled".equals(entry.getKey())) {
                             ignoreThrottled = value;
-                        } else if (parserConfig.restApiVersion() == RestApiVersion.V_7
-                            && ("type".equals(entry.getKey()) || "types".equals(entry.getKey()))) {
-                                deprecationLogger.compatibleCritical("msearch_with_types", RestMultiSearchAction.TYPES_DEPRECATION_MESSAGE);
```
[the remark this answers is not in the document]
Comment (drempapis) [reply server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java:286]: I removed it :)

File: server/src/main/java/org/elasticsearch/action/termvectors/TermVectorsRequest.java [hunk 3]
```diff
@@ -583,8 +581,6 @@ public static void parseRequest(TermVectorsRequest termVectorsRequest, XContentP
                     termVectorsRequest.version = parser.longValue();
                 } else if (VERSION_TYPE.match(currentFieldName, parser.getDeprecationHandler())) {
                     termVectorsRequest.versionType = VersionType.fromString(parser.text());
-                } else if (restApiVersion == RestApiVersion.V_7 && TYPE.match(currentFieldName, parser.getDeprecationHandler())) {
-                    deprecationLogger.compatibleCritical("termvectors_with_types", RestTermVectorsAction.TYPES_DEPRECATION_MESSAGE);
```
[the remark this answers is not in the document]
Comment (drempapis) [reply server/src/main/java/org/elasticsearch/action/termvectors/TermVectorsRequest.java:587]: I removed the constant `RestTermVectorsAction.TYPES_DEPRECATION_MESSAGE`, but I am not sure what you mean by "usages"

[the remark this answers is not in the document]
Comment (javanna) [reply server/src/main/java/org/elasticsearch/action/termvectors/TermVectorsRequest.java:587]: we are good. I was looking at the 8.x branch when I commented, that has additional usages. Sorry for the noise.

**Review by javanna — APPROVED** [review]
LGTM thanks!

**Pull request merged**